init.go

```diff
@@ -10,91 +10,128 @@ import (
+	initState.SetBytes(common.CoreIDKey, []byte(newCore.GetObjectID()))
+
+	newImpulse := mustNewImpulse(config)
+	initState.SetImpulse(newImpulse)
+	initState.SetBytes(common.ImpulseIDKey, []byte(newImpulse.GetObjectID()))
+
+	newNetwork := mustNewNetwork(config)
+	initState.SetNetwork(newNetwork)
+	initState.SetBytes(common.NetworkIDKey, []byte(newNetwork.GetObjectID()))
+
+	newCharacterNeuron := mustNewCharacterNeuron(config)
+	initState.SetNeuron(newCharacterNeuron)
+	initState.SetBytes(common.CharacterNeuronIDKey, []byte(newCharacterNeuron.GetObjectID()))
+
+	newFirstNeuron := mustNewFirstNeuron(config)
+	initState.SetNeuron(newFirstNeuron)
+	initState.SetBytes(common.FirstNeuronIDKey, []byte(newFirstNeuron.GetObjectID()))
+
+	newJobNeuron := mustNewJobNeuron(config)
+	initState.SetNeuron(newJobNeuron)
+	initState.SetBytes(common.JobNeuronIDKey, []byte(newJobNeuron.GetObjectID()))
 
-	return initPair{Object: newNetwork, State: newNetworkState}
+	object.SetState(common.InitStateKey, initState)
+
+	return object
 }
 
-func mustGetCoreInitPair(config initPairConfig) initPair {
-	newCoreConfig := core.DefaultConfig()
-	newCoreConfig.Log = config.Log
-	newCoreConfig.TextGateway = config.TextGateway
-	newCore := core.NewCore(newCoreConfig)
-	newCoreState, err := newCore.GetState(common.DefaultStateKey)
+func mustPrepareCoreInitState(config initConfig, core spec.Core) spec.Core {
+	defaultState, err := core.GetState(common.DefaultStateKey)
 	if err != nil {
 		panic(err)
 	}
-	newCoreState = newCoreState.Copy()
+	initState := defaultState.Copy()
+
+	newCore := common.MustObjectToCore(mustPrepareInitState(config, mustNewCore(config)))
+	initState.SetCore(newCore)
+	initState.SetBytes(common.CoreIDKey, []byte(newCore.GetObjectID()))
+
+	newImpulse := common.MustObjectToImpulse(mustPrepareInitState(config, mustNewImpulse(config)))
+	initState.SetImpulse(newImpulse)
+	initState.SetBytes(common.ImpulseIDKey, []byte(newImpulse.GetObjectID()))
+
+	newNetwork := common.MustObjectToNetwork(mustPrepareInitState(config, mustNewNetwork(config)))
+	initState.SetNetwork(newNetwork)
+	initState.SetBytes(common.NetworkIDKey, []byte(newNetwork.GetObjectID()))
+
+	newCharacterNeuron := common.MustObjectToNeuron(mustPrepareInitState(config, mustNewCharacterNeuron(config)))
+	initState.SetNeuron(newCharacterNeuron)
+	initState.SetBytes(common.CharacterNeuronIDKey, []byte(newCharacterNeuron.GetObjectID()))
+
+	newFirstNeuron := common.MustObjectToNeuron(mustPrepareInitState(config, mustNewFirstNeuron(config)))
+	initState.SetNeuron(newFirstNeuron)
+	initState.SetBytes(common.FirstNeuronIDKey, []byte(newFirstNeuron.GetObjectID()))
+
+	newJobNeuron := common.MustObjectToNeuron(mustPrepareInitState(config, mustNewJobNeuron(config)))
+	initState.SetNeuron(newJobNeuron)
+	initState.SetBytes(common.JobNeuronIDKey, []byte(newJobNeuron.GetObjectID()))
+
+	core.SetState(common.InitStateKey, initState)
 
-	return initPair{Object: newCore, State: newCoreState}
+	return core
```

Comment:
All this can be simplified. The state chain and object factory mechanism needs to be implemented as designed in the concept docs.